src/components/DataListSection.tsx

```diff
@@ -64,16 +64,23 @@ export const DataListSection = ({
   showButton,
   showLink
 }: Props) => {
-  if (loading) {
-    return (
-      <View>
-        {!!sectionTitle && <SectionHeader onPress={navigate} title={sectionTitle} />}
+  const renderSectionHeader = () => {
+    if (!sectionTitle) return null;
 
-        <LoadingContainer>
-          <ActivityIndicator color={colors.refreshControl} />
-        </LoadingContainer>
-      </View>
-    );
+    return <SectionHeader onPress={navigate} title={sectionTitle} />;
+  };
+
+  const renderLoadingState = () => (
+    <View>
+      {renderSectionHeader()}
+      <LoadingContainer>
+        <ActivityIndicator color={colors.refreshControl} />
+      </LoadingContainer>
+    </View>
+  );
+
+  if (loading) {
+    return renderLoadingState();
```

Comment:
i can understand the refactoring of `renderSectionHeader` to avoid `&&` in rendering, but how is this better than before?

```
if (loading) {
  return (
    <View>
      {renderSectionHeader()}
      <LoadingContainer>
        <ActivityIndicator color={colors.refreshControl} />
      </LoadingContainer>
    </View>
  );
}
```

Reply:
With `renderSectionHeader`, code repetition is avoided. As I noticed, the same code was used in 3 different sections, now only `renderSectionHeader` is used. Adding the if check instead of the `&&` operator solves the complexity of the code and I think it is more understandable.

Reply:
i think this does not answer my question. as is said, the refactoring of `renderSectionHeader` is clear to me.
but i see no reason for a method `renderLoadingState` that has benefits over my postet code before.

Reply:
Yes, that's right. I removed `renderLoadingState` from the code because it doesn't have any extra benefit.